ArjunaJTA/jta/classes/com/arjuna/ats/internal/jta/transaction/arjunacore/jca/TransactionImporterImple.java

```diff
@@ -211,17 +238,33 @@ public void removeImportedTransaction(Xid xid) throws XAException
 	}
 	
 	public Set<Xid> getInflightXids(String parentNodeName) {
-		Iterator<TransactionImple> iterator = _transactions.values().iterator();
+		Iterator<TransactionImpleHolder> iterator = _transactions.values().iterator();
 		Set<Xid> toReturn = new HashSet<Xid>();
 		while (iterator.hasNext()) {
-			TransactionImple next = iterator.next();
-			if (next.getParentNodeName().equals(parentNodeName)) {
-				toReturn.add(next.baseXid());
+			TransactionImpleHolder next = iterator.next();
+			TransactionImple imported = null;
+			try {
+				imported = next.getImported();
+			} catch (InterruptedException e) {
+				throw new RuntimeException(e);
```

Comment:
You need to re-set the interrupted status of the thread here by calling ```Thread.currentThread().interrupt()```.  Also, wrapping InterruptedException is generally not a good idea; better to just have a message that says "operation was interrupted" or something.